contracts/TwabRewards.sol

```diff
@@ -0,0 +1,451 @@
+
+import "./interfaces/ITwabRewards.sol";
+
+/**
+ * @title PoolTogether V4 TwabRewards
+ * @author PoolTogether Inc Team
+ * @notice Contract to distribute rewards to depositors in a pool.
+ */
+contract TwabRewards is ITwabRewards, Manageable {
+    using SafeERC20 for IERC20;
+
+    /* ============ Global Variables ============ */
+
+    /// @notice Settings of each promotion.
+    mapping(uint256 => Promotion) internal _promotions;
+
+    /// @notice Latest promotion id.
+    uint256 internal _latestPromotionId;
+
+    /// @notice Keeps track of claimed rewards per user.
+    /// @dev _claimedEpochs[promotionId][user] => claimedEpochs
+    /// @dev We pack epochs claimed by a user into a uint256. So we can't store more than 256 epochs.
+    mapping(uint256 => mapping(address => uint256)) internal _claimedEpochs;
+
+    /* ============ Events ============ */
+
+    /**
+        @notice Emitted when contract has been deployed.
+        @param owner Contract owner address
+   */
+    event Deployed(address owner);
+
+    /**
+        @notice Emmited when a promotion is created.
+        @param id Id of the newly created promotion
+    */
+    event PromotionCreated(uint256 id);
+
+    /**
+        @notice Emmited when a promotion is extended.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the rewards contract
+    */
+    event PromotionCancelled(IERC20 token, uint256 amount);
+
+    /**
+        @notice Emmited when a promotion is extended.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the recipient address
+        @param numberOfEpochs New number of epochs after extending the promotion
+    */
+    event PromotionExtended(IERC20 token, uint256 amount, uint256 numberOfEpochs);
+
+    /**
+        @notice Emmited when rewards have been claimed.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the recipient address
+    */
+    event RewardsClaimed(IERC20 token, uint256 amount);
+
+    /* ============ Constructor ============ */
+
+    /**
+        @notice Deploy TwabRewards contract.
+        @param _owner Contract owner address
+    */
+    constructor(address _owner) Ownable(_owner) {
+        emit Deployed(_owner);
+    }
+
+    /* ============ Modifiers ============ */
+
+    /// @dev Ensure that the caller is the creator of the currently active promotion.
+    /// @param _promotionId Id of the promotion to check
+    modifier onlyPromotionCreator(uint256 _promotionId) {
+        require(
+            msg.sender == _getPromotion(_promotionId).creator,
+            "TwabRewards/only-promotion-creator"
+        );
+        _;
+    }
+
+    /* ============ External Functions ============ */
+
+    /// @inheritdoc ITwabRewards
+    function createPromotion(address _ticket, PromotionParameters calldata _promotionParameters)
```

Comment:
Definitely just have all the params here- having a ticket then separate struct is odd.

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/bc5f7b27d921335a65cc47925270cb37dcca3213